### ISSUE TITLE
Problem of Evaluation Protocols

### ISSUE THREAD

**Issue by Ocxs** [HKUDS/DCRec#8]
In Section 3.1.2 of your paper, you mentioned, "We follow [11, 20, 30] to adopt the leave-one-out strategy for model evaluation. Specifically, we treat the last interaction of each user as testing data, and designate the previous one as validation data. " However, the evaluation protocol employed in your released code does not correspond to the leave-one-out strategy. The log information for DCRec is as follows:
```
eval_args = {'mode': 'pop100', 'order': 'TO', 'split': {'RS': [0.8, 0.1, 0.1]}, 'group_by': 'user'}
```
In RecBole, the 'RS' parameter divides the dataset into train, val, and test sections based on ratios.

<img width="777" alt="image" src="https://github.com/HKUDS/DCRec/assets/11175797/c929274c-800a-4a2c-b49e-66a5ac955b4c">

**Comment by yuh-yang** [comment]
Hi,

The automatic splitting function provided by RecBole would be deactivated if preprocessed data files are available. Please refer to run_DCRec.py line 305-320.

**Comment by Ocxs** [comment]
<img width="997" alt="image" src="https://github.com/HKUDS/DCRec/assets/11175797/e7bbadd3-8962-47ca-bbe5-d632952cb2db">

Thank you for your response. Thus, the test set utilizes the leave-one-out strategy, yet when dividing the validation set, the split_by_ratio method is employed. Default parameter settings dictate that 90% of the dataset forms the training set, with the remaining 10% functioning as the validation set. Does my comprehension hold true?

**Comment by Ocxs** [comment]
<img width="1066" alt="image" src="https://github.com/HKUDS/DCRec/assets/11175797/837fe748-b638-4eb4-9941-0c7b50947676">
By looking at this code piece, it seems like if I set args.validation to True, the validation set is used as the test set. And I haven't seen any actual test set data being used for evaluation before the code is finished running. The results on the validation set seem pretty close to the ones mentioned in your paper. Could this be a problem?

**Comment by yuh-yang** [comment]
Hi,

validation is used for selecting best checkpoint. You can then test your best ckpt with the true test set to have finalized performance results.

**Comment by yuh-yang** [comment]
Closed due to inactiveness. Reopen it if you have further questions.
